buildSrc/src/main/java/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -108,10 +109,18 @@ public void apply(Project project) {
         TaskProvider<Copy> copyUpgradeTask = configureCopyUpgradeTask(project, upgradeVersion, upgradeDir);
         TaskProvider<Copy> copyPluginsTask = configureCopyPluginsTask(project, pluginsDir);
 
+        // Create lifecycle tasks so we can run all tests of a given type
+        Map<ElasticsearchDistribution.Type, TaskProvider<?>> lifecyleTasks = new HashMap<>();
+        lifecyleTasks.put(Type.DOCKER, project.getTasks().register("destructiveDistroTest#docker"));
+        lifecyleTasks.put(Type.ARCHIVE, project.getTasks().register("destructiveDistroTest#archives"));
+        lifecyleTasks.put(Type.DEB, project.getTasks().register("destructiveDistroTest#packages"));
+        lifecyleTasks.put(Type.RPM, lifecyleTasks.get(Type.DEB));
+
         TaskProvider<Task> destructiveDistroTest = project.getTasks().register("destructiveDistroTest");
         for (ElasticsearchDistribution distribution : distributions) {
             TaskProvider<?> destructiveTask = configureDistroTest(project, distribution, dockerSupport);
             destructiveDistroTest.configure(t -> t.dependsOn(destructiveTask));
+            Optional.ofNullable(lifecyleTasks.get(distribution.getType())).ifPresent(p -> p.configure(t -> t.dependsOn(destructiveTask)));
```

Comment:
When would this not be present? Shouldn't that be an error?

Reply:
Yeah, you're right. I had initially anticipated something like the integ test zip type, but we aren't testing that here. And if we _do_ add a type we would probably want this to explode rather than silently ignore. I've removed the optional wrapper here.